### ISSUE TITLE
Wrong pyf with declaration of parameters in modules

### ISSUE THREAD

**Issue by ndarmage** [numpy/numpy#8730]
f2py provided with numpy 1.12.0 produces wrong pyf signature in case of parameter declaration in modules. Noticed with python2.7.11. Calling f2py by python3.4 doesn't show this issue.

write the following in file.f90:
```
module ppcolloc_data
  implicit none
  integer ( kind = 4 ), parameter :: npiece = 100
  integer ( kind = 4 ), parameter :: ncoef = 2000
  ! old common approx
  integer ( kind = 4 ) l, kpm
  real ( kind = 8 ), dimension(npiece) :: break
  real ( kind = 8 ), dimension(ncoef) :: coef
  save
end module ppcolloc_data
```
f2py file.f90 -m file -h file.pyf

Note the output to screen:
```
            Block: ppcolloc_data
In: :file:file.f90:ppcolloc_data
get_parameters: got "unexpected EOF while parsing (<string>, line 0)" on ''
In: :file:file.f90:ppcolloc_data
get_parameters: got "unexpected EOF while parsing (<string>, line 0)" on ''
```

head file.pyf
shows:
```
!    -*- f90 -*-
! Note: the context of this file is case sensitive.

python module file ! in 
    interface  ! in :file
        module ppcolloc_data ! in :file:file.f90
            real(kind=8) dimension() :: break_bn
            real(kind=8) dimension() :: coef
            integer(kind=4), parameter,optional :: ncoef=2000
            integer(kind=4), parameter,optional :: npiece=100
```
Note that vector dimensions are missing.

lines like:
character (len = 2), parameter, dimension(3) :: x=(/ 'ab', 'bc, 'cd' /)
cannot be treated too, with both py2.7 and py3.4.

Thanks in advance for your support.

**Comment by pdebuyl** [comment]
Using Python 3 and
```
Version:     2
numpy Version: 1.16.1
```

I cannot reproduce the first error.

Parsing ` character (len = 2), parameter, dimension(3) :: x=(/ 'ab', 'bc, 'cd' /)` fails but I am not sure that this is supported by f2py anyway.

**Comment by melissawm** [comment]
Confirming - I cannot reproduce on `main` and the parsing of array parameters should be fixed by #15457

**Comment by pdebuyl** [comment]
Just passing by. NumPy 1.22 and Python 3.9 on linux amd64. Cannot reproduce.

**Comment by HaoZeke** [comment]
Closing as fixed, for `1.22.3`:

```
!    -*- f90 -*-
! Note: the context of this file is case sensitive.

python module file ! in 
    interface  ! in :file
        module ppcolloc_data ! in :file:file.f90
            integer(kind=4), parameter,optional :: npiece=100
            integer(kind=4), parameter,optional :: ncoef=2000
            integer(kind=4) :: l
            integer(kind=4) :: kpm
            real(kind=8) dimension(100) :: break_bn
            real(kind=8) dimension(2000) :: coef
        end module ppcolloc_data
    end interface 
end python module file

! This file was auto-generated with f2py (version:1.22.3).
! See:
! https://web.archive.org/web/20140822061353/http://cens.ioc.ee/projects/f2py2e
```
